cfme/automate/buttons.py

```diff
@@ -573,6 +582,7 @@ class ButtonGroupCollection(BaseCollection):
     VM_INSTANCE = "VM and Instance"
 
     def instantiate(self, text, hover, type, image=None, display=None, icon_color=None):
+        self.type = type
```

Comment:
ahhhh; This is a bad implementation but yes, to provide a type to collection class; we have to assign it like this.
you can check `create` a method https://github.com/ManageIQ/integration_tests/blob/master/cfme/automate/buttons.py#L585

I am planning to change the structure of `button.py` but for now let's go with it.

Reply:
http://pastebin.test.redhat.com/655607

a collection should know object type as per model implementation.
1. when I created object > collection know type is `Provider` so everything is ok
2. but If I instantiate any other object with different type > collection does not know about new object type. to avoid this I am trying to provide new object type.

I know this is bad for now. it's the only solution; I found to avoid this situation.